l1-contracts/src/core/libraries/Decoder.sol

```diff
@@ -346,70 +349,100 @@ library Decoder {
+        assembly {
+          calldatacopy(dstPtr, add(_l2Block.offset, mload(add(offsets, 0x40))), bytesToCopy)
+        }
+        dstPtr += bytesToCopy;
 
-          // Adding new l2 to l1 msgs
-          calldatacopy(dstPtr, add(_l2Block.offset, mload(add(offsets, 0x60))), mul(0x04, 0x20))
-          dstPtr := add(dstPtr, mul(0x04, 0x20))
+        // Adding new l2 to l1 msgs
+        bytesToCopy = Constants.L2_TO_L1_MSGS_NUM_BYTES_PER_BASE;
+        assembly {
+          calldatacopy(dstPtr, add(_l2Block.offset, mload(add(offsets, 0x60))), bytesToCopy)
+        }
+        dstPtr += bytesToCopy;
 
-          // Adding Contract Leafs
-          calldatacopy(dstPtr, add(_l2Block.offset, mload(add(offsets, 0x80))), mul(0x2, 0x20))
-          dstPtr := add(dstPtr, mul(0x2, 0x20))
+        // Adding Contract Leafs
+        bytesToCopy = Constants.CONTRACTS_NUM_BYTES_PER_BASE;
+        assembly {
+          calldatacopy(dstPtr, add(_l2Block.offset, mload(add(offsets, 0x80))), bytesToCopy)
+        }
+        dstPtr += bytesToCopy;
 
-          // Kernel1.contract.aztecAddress
-          let contractDataOffset := mload(add(offsets, 0xa0))
+        // Kernel1.contract.aztecAddress
+        uint256 contractDataOffset = offsets.contractDataOffset;
+        assembly {
           calldatacopy(dstPtr, add(_l2Block.offset, contractDataOffset), 0x20)
-          dstPtr := add(dstPtr, 0x20)
+        }
+        dstPtr += 0x20;
 
-          // Kernel1.contract.ethAddress padded to 32 bytes
-          // Add 12 (0xc) bytes of padding to the ethAddress
-          dstPtr := add(dstPtr, 0xc)
+        // Kernel1.contract.ethAddress padded to 32 bytes
+        // Add 12 (0xc) bytes of padding to the ethAddress
+        dstPtr += 0xc;
+        assembly {
           calldatacopy(dstPtr, add(_l2Block.offset, add(contractDataOffset, 0x20)), 0x14)
-          dstPtr := add(dstPtr, 0x14)
+        }
+        dstPtr += 0x14;
 
-          // Kernel2.contract.aztecAddress
+        // Kernel2.contract.aztecAddress
+        assembly {
           calldatacopy(dstPtr, add(_l2Block.offset, add(contractDataOffset, 0x34)), 0x20)
-          dstPtr := add(dstPtr, 0x20)
+        }
+        dstPtr += 0x20;
 
-          // Kernel2.contract.ethAddress padded to 32 bytes
-          // Add 12 (0xc) bytes of padding to the ethAddress
-          dstPtr := add(dstPtr, 0xc)
+        // Kernel2.contract.ethAddress padded to 32 bytes
+        // Add 12 (0xc) bytes of padding to the ethAddress
+        dstPtr += 0xc;
+        assembly {
           calldatacopy(dstPtr, add(_l2Block.offset, add(contractDataOffset, 0x54)), 0x14)
+        }
 
-          // encryptedLogsHashKernel1
-          dstPtr := add(dstPtr, 0x14)
-          mstore(dstPtr, mload(add(vars, 0x60))) // `encryptedLogsHashKernel1` starts at 0x60 in `vars`
+        // encryptedLogsHashKernel1
+        dstPtr += 0x14;
+        bytes32 hash = vars.encrypedLogsHashKernel1;
```

Comment:
If the ordering of `encryptedLogsHashKernel1` etc are changed in the `ConsumablesVars` you should be able to load in all 4 at once with `identity` instead of having to jump through one at a time.

Reply:
ah very true